src/components/ManageWallets/PrimaryWalletRow.tsx

```diff
@@ -0,0 +1,58 @@
+import styled from 'styled-components';
+
+import { Button } from '../core/Button/Button';
+import colors from '../core/colors';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseM, BODY_MONO_FONT_FAMILY, TitleDiatypeM } from '../core/Text/Text';
+import { NewTooltip } from '../Tooltip/NewTooltip';
+import { useTooltipHover } from '../Tooltip/useTooltipHover';
+import { walletIconMap } from '../WalletSelector/multichain/WalletButton';
+
+type PrimaryWalletProps = {
+  address: string | null;
+  chain: string;
+};
+
+export default function PrimaryWalletRow({ address, chain }: PrimaryWalletProps) {
+  const iconUrl = walletIconMap[chain.toLowerCase() as keyof typeof walletIconMap];
+
+  const { floating, reference, getFloatingProps, getReferenceProps, floatingStyle } =
+    useTooltipHover();
+
+  return (
+    <PrimaryWallet gap={12}>
+      <TitleDiatypeM>Primary</TitleDiatypeM>
+      <HStack align="center" justify="space-between">
+        <HStack gap={12} align="center">
+          <Icon src={iconUrl} />
+          <StyledWalletAddress>{address}</StyledWalletAddress>
+        </HStack>
+        <div {...getReferenceProps()} ref={reference}>
+          <Button variant="warning" disabled>
+            Disconnect
+          </Button>
+        </div>
+        <NewTooltip
```

Comment:
trying out @tbezman 's new tooltip

Reply:
i think we can make this API more ergo. thinking we can combine the `style` and `ref` into `getFloatingProps` somehow